resources/policies/ingress_modsec_nginx_class.rego

```diff
@@ -0,0 +1,11 @@
+package cloud_platform.admission
+
+# This policy deny any ingress that have "kubernetes.io/ingress.class: nginx" annotation(using default ingress-controller) 
+# and use 'nginx.ingress.kubernetes.io/enable-modsecurity: "true"' annotation.
+
+deny[msg] {
+  input.request.kind.kind == "Ingress"
+  input.request.object.metadata.annotations["kubernetes.io/ingress.class"] == "nginx"
+  input.request.object.metadata.annotations["nginx.ingress.kubernetes.io/enable-modsecurity"] == "true"
```

Comment:
@vijay-veeranki , should we check if the annotation is "true" or just deny if that annotation is present?

Reply:
I would say check for "true" because if someone has set that annotation to "false" we're happy. Having said that, I don't know what would happen if they set it to "cheese" or whatever, but I think testing for "true" is easier to understand than explicitly allowing "false" and rejecting any other values.